prow/plugins/label/label.go

```diff
@@ -226,19 +230,28 @@ func handle(gc githubClient, log *logrus.Entry, ae assignEvent) error {
 	}
 
 	toRepeat := []string{}
+	isMember := false
 	if len(sigMatches) > 0 {
-		isMember, _ := gc.IsMember(ae.org, ae.login)
-		if !isMember {
-			toRepeat = ae.getRepeats(sigMatches, existingLabels)
-		}
+		isMember, _ = gc.IsMember(ae.org, ae.login)
+		toRepeat = ae.getRepeats(sigMatches, existingLabels)
 	}
 
 	if len(toRepeat) > 0 {
 		msg := fmt.Sprintf(chatBack, strings.Join(toRepeat, ", "))
-		if err := gc.CreateComment(ae.org, ae.repo, ae.number, plugins.FormatResponseRaw(ae.body, ae.url, ae.login, msg)); err != nil {
-			log.WithError(err).Errorf("Could not create comment \"%s\".", msg)
+		if !isMember {
+			if err := gc.CreateComment(ae.org, ae.repo, ae.number, plugins.FormatResponseRaw(ae.body, ae.url, ae.login, msg)); err != nil {
+				log.WithError(err).Errorf("Could not create comment \"%s\".", msg)
+			}
 		}
 
+		if sc != nil {
+			//if sig matches then send a notification on slack
+			for _, sig := range sigMatches {
+				if err := sc.WriteMessage(plugins.FormatResponseRaw(ae.body, ae.url, ae.login, msg), "sig-"+sig[1]); err != nil {
```

Comment:
It might be better to post a different message to the slack channel instead of using the `chatBack` message format string since that only really makes sense in the context of reiterating a mention on github. Maybe define `msg` inside both the `if len(toRepeat) > 0` and `if sc != nil` conditionals with different values.